miio/discovery.py

```diff
@@ -17,9 +17,9 @@
     "roborock-vacuum-s5": Vacuum,
     "chuangmi-plug-m1": Plug,
     "chuangmi-plug-v2": Plug,
-    "chuangmi-plug-v1": PlugV1,
-    "chuangmi-plug_": PlugV1,
-    "chuangmi-plug-v3": PlugV3,
+    "chuangmi-plug-v1": ChuangmiPlug,
+    "chuangmi-plug_": ChuangmiPlug,
+    "chuangmi-plug-v3": ChuangmiPlug,
```

Comment:
For passing a parameter here you can use `functools.partial`, so something like:
```
partial(ChuangmiPlug, model=MyModel)
```
can be used in this case.

I'm wondering whether that should be an enum, or simply an integer `model_version`, which would default to either of them? We could also derive the model (if not given) per default from the status information (to adapt the setters accordingly), as it's just one extra attribute & different naming?

Reply:
I will care about tomorrow.

Reply:
The `check_and_create_device` method doesn't enter the `inspect.isclass(v)` branch anymore if I use a partial. I'm unsure about a good implementation.

Reply:
Indeed.. Hmm, maybe adding a check if it's partial and extracting the class out from it?
```
>>> from functools import partial
>>> class A:
>>>     pass

>>> partial(A, foo=1).func
<class '__main__.A'>
```